dspace-api/src/main/java/org/dspace/versioning/VersioningServiceImpl.java

```diff
@@ -170,6 +143,51 @@ public void removeVersion(Context c, Item item) throws SQLException {
         }
     }
 
+    @Override
+    public void removeVersionFromItem(Context c, Item item) {
+        try{
+            Version version = versionDAO.findByItem(c, item);
+            if(version!=null){
+                // we will first delete the version and then the item
+                // after deletion of the version we cannot find the item anymore
+                // so we need to get the item now
+                deleteVersion(c, item, version);
+            }
+        }catch (Exception e) {
+            c.abort();
```

Comment:
This `abort()` worries me slightly.  Should we really be allowing a method like this to abort *all* changes on the current `Context`, or is that a decision to be made at a level above (based on whether an exception is thrown).  It's possible this `abort()` isn't even necessary, since throwing an exception *should* already result in changes not being committed.